ui_framework/components/button/button.js

```diff
@@ -0,0 +1,147 @@
+import React, {
+  PropTypes,
+} from 'react';
+
+import classNames from 'classnames';
+import keyMirror from 'keymirror';
+
+import { KuiButtonIcon } from './button_icon/button_icon';
+
+const commonPropTypes = {
+  type: PropTypes.string,
```

Comment:
fwiw I prefer using the enum style for these things, instead of the plain strings.  I think it gives an obvious single location where all the types we handle are defined, and also indicates that the string is not meant for ui consumption.